### ISSUE TITLE
Multipart/form-data support for file uploading

### ISSUE THREAD

**Issue by jenshaase** [finagle/finch#197]
Hi,

If I'm correct Finagle currently doesn't support `multipart/form-data` out of the box. However, Finatra (http://finatra.info/docs/index.html#uploads) has build in support for it.

It would be nice if there is the possibility to upload a file within finch.

Cheers,
Jens

**Comment by jenshaase** [comment]
This code may be a solution (adopted from the finatra code):

~~~
  import com.twitter.finagle.httpx.netty.Bijections._
  import scala.collection.JavaConversions._
  import org.jboss.netty.handler.codec.http.multipart.{HttpPostRequestDecoder, MixedAttribute, MixedFileUpload}

  object OptionalMultipartParam {
    def apply(param: String): RequestReader[Option[Array[Byte]]] =
      RequestReader(items.ParamItem(param)) { req =>
        val dec = new HttpPostRequestDecoder(from(req))
        dec.getBodyHttpDatas.find(_.getName == param).flatMap{
          case data: MixedFileUpload => Some(data.get)
          case data: MixedAttribute => Some(data.get)
          case _ => None
        }
      }
  }

  object RequiredMultipartParam {
    def apply(param: String) =
      OptionalMultipartParam(param).failIfEmpty.shouldNot("be empty")(_.isEmpty)
  }
~~~

**Comment by vkostyukov** [comment]
Interesting. Thanks @jenshaase for failing a ticket. Do you mind to create a pull-request? It would be fair if you implement this since it's your idea/code. This is an amazing feature.

**Comment by jenshaase** [comment]
Ok. I can submit a pull request in the next days.

**Comment by vkostyukov** [comment]
Cool! Thanks again!

**Comment by vkostyukov** [comment]
Fixed by #201.
